### ISSUE TITLE
PMD is executed for all file types regardless of active rules

### ISSUE THREAD

**Issue by adangel** [pmd/pmd-eclipse-plugin#88]
**Description**: Given a ruleset, that only contains rules for Java, the pmd-eclipse-plugin executes PMD not only on Java files, but on all resources in the project.
This not only is a performance problem, but this leads also to unrelated errors (e.g. a JavaScript file that cannot be parsed).

**Expected Result:**
The plugin should detect the applicable file types based on the selected rules. E.g. only *.java rules should be executed.
The command line interface of PMD is doing this already.

**Workaround:**
Using a file-filter, you can exclude problematic files:

```xml
<exclude-pattern>.*\.js</exclude-pattern>
```

Since the includes are overriding the excludes, you can exclude everything except for java files:

```xml
   <exclude-pattern>.*</exclude-pattern>
   <include-pattern>.*\.java</include-pattern>
```

Documentation: https://pmd.github.io/pmd-6.11.0/pmd_userdocs_making_rulesets.html#filtering-the-processed-files

This can also be configured via Window, Preferences, PMD, File Filter:
![grafik](https://user-images.githubusercontent.com/1573684/52413984-5005a200-2ae3-11e9-9ef8-93325e5348f8.png)

**Comment by adangel** [comment]
TODO:

- [ ] Add new configuration option: "Determine applicable file types automatically" under PMD preferences
- [ ] Determine all languages used in the ruleset (similar to PMD.getApplicableLanguages())
- [ ] Adjust BaseVisitor.reviewResource() to only execute PMD for files with matching extension
